Get to a state where customers can test and give feedback

murilommen opened the pull request:
On this PR I have focused on making the toolkit more usable and reliable. To do so, I have:
- extended the `monitor/manager/README.md` examples and useful commands
- created unit tests
- made env vars definable before OR after importing modules
- simplified imports
- other refactors with internal usage feedbacks

murilommen committed:
including TODOs and docs refactors

murilommen committed:
addressing feedback + test shells

* put desired unit tests on a package
* make import not rely on env var
* add exclude columns method

murilommen committed:
improved documentation and usage

murilommen committed:
test builder, manager and credentials

murilommen committed:
adding more config examples

murilommen committed:
fix format

murilommen committed:
fix broken test

murilommen committed:
validating columns input

murilommen committed:
using columns that exist to pass validation

murilommen committed:
minor refactor on test inheritance

murilommen committed:
fixing actions adding behaviour - lacks testing

murilommen committed:
adding unit tests to notification actions

naddeoa commented on tests/monitor/manager/test_builder.py:
```diff
@@ -0,0 +1,90 @@
+    assert not builder.monitor
+    assert not builder.analyzer
+
+    builder.build()
+
+    assert isinstance(builder.monitor, Monitor)
+    assert isinstance(builder.analyzer, Analyzer)
```
This part seems a little suspect to me. Does this mean that `builder.build()` has a side effect that changes `builder.monitor`? It would be best if it was just a function that returned something new if thats the case.

murilommen replied:
so my idea is that `build()` will be the thing to create or modify the existing `builder.monitor`. wdym by returning something new?

murilommen replied:
calling it `MonitorSetup` now! lmk if this works

naddeoa commented on whylabs_toolkit/helpers/monitor_helpers.py:
```diff
@@ -19,7 +20,11 @@ def get_monitor_config(org_id: str, dataset_id: str) -> Any:
 
-def get_monitor(org_id: str, dataset_id: str, monitor_id: str) -> Any:
+def get_monitor(monitor_id: str, org_id: Optional[str], dataset_id: Optional[str]) -> Any:
+    if not org_id:
+        org_id = Config().get_default_org_id()
+    if not dataset_id:
+        dataset_id = Config().get_default_dataset_id()
```
there can be defaults for these? I thought we would have to throw.

murilommen replied:
put this as "either pass as an argument or get it from the default env var"

naddeoa commented on whylabs_toolkit/helpers/utils.py:
```diff
@@ -1,12 +1,29 @@
+    elif isinstance(action, EmailRecipient):
+        return "email"
+    else:
+        raise ValueError(f"Can't work with {action} type")
```
Is this something the user can fix on their own when it happens? If not then we should say that its a bug in the library so they have some hint of what to do when this happens. Same applies to other exceptions.

murilommen replied:
what would they fix? this is just a helper method to be able to use `PUT` correctly for each action. thing is that they are only allowed to auto-put with email or slack. other stuff they will need to 1. create it first and 2. use global actions. 
maybe the best is that this is just a private static method built inside of the Monitor Manager?

murilommen committed:
changing name to make functionality clearer

MonitorBuilder wasn't actually implementing the Builder pattern.
Changing that to MonitorSetup to make it clearer
